### PR TITLE
Tanisha New Units

### PR DESCRIPTION
**Pull request by tanishagarg2**

### CODE REVIEW

File: src/assets/Jalapeno.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	public void attack(Board board) {
+		int row = getRow();
+		int column = getCol();
+		
+		ArrayList<Zombie> zombieInRow = new ArrayList<>();
+		zombieInRow = (ArrayList<Zombie>) board.getRowTargets(row, column);
+		
+		for(Zombie z : zombieInRow) {
+			z.setHp(0);
+			board.removeZombie(z.getRow(), z.getCol());
```
Comment (DwangDistracted) [inline src/assets/Jalapeno.java]: Are we sure that this won't cause a ConcurrentModificationException

File: src/assets/PlantTypes.java [hunk 2]
```diff
@@ -9,14 +9,17 @@
  */
 public enum PlantTypes {
 	SUNFLOWER,
+	TWIN_SUNFLOWER,
```
Comment (DwangDistracted) [inline src/assets/PlantTypes.java]: These types need to be added to the toPlant() Method

File: src/assets/Repeater_Peashooter.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	public PlantTypes getPlantType() {
+		return PLANT_TYPE;
+	}
+	
+	@Override
+	public void attack(Board board) {
+		int row = getRow();
+		int column = getCol();
+		
+		Zombie zombieTarget = board.getSingleZombieTarget(row, column);
```
Comment (DwangDistracted) [inline src/assets/Repeater_Peashooter.java]: This plant should be attacking the row

File: src/assets/Repeater_Peashooter.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
+package assets;
+
+import engine.Board;
+import util.Logger;
+
+public class Repeater_Peashooter extends Plant{
+private static Logger LOG = new Logger("Repeated Peashooter");
+	
+	private static final int DEFAULT_HP = HEALTH_MEDIUM;
+	private static final int DEFAULT_POWER = ATTACK_LOW + ATTACK_LOW;
```
Comment (DwangDistracted) [inline src/assets/Repeater_Peashooter.java]: shouldn't this be attack low?

File: src/assets/Twin_Flower.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+import engine.Board;
+
+public class Twin_Flower extends Plant{
+	private static final int DEFAULT_HP = HEALTH_LOW;
+	private static final int DEFAULT_POWER = ATTACK_NONE;
+	private static final int COST = 50;
+	
+	//The Amount of Points that are added per turn per sunflower
+	private static final int POINTS = 10; //points that a sunflower provides a player to buy other plants
```
Comment (DwangDistracted) [inline src/assets/Twin_Flower.java]: This should be higher. 10 is the amount that a single sunflower gives

File: src/assets/Vaulting_Zombie.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+	}
+	
+	/**
+	 * Zombie attack method.
+	 */
+	@Override
+	public void attack(Board board) {
+		Plant plantTarget = board.getPlant(getRow(), getCol());
+		if(this.getCol()==plantTarget.getCol()+2) {
+			this.setDefaultSpeed();
```
Comment (DwangDistracted) [inline src/assets/Vaulting_Zombie.java]: What is this doing here?

File: src/assets/Vaulting_Zombie.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+		this.default_speed = SPEED_LOW;
+	}
+	
+	/**
+	 * Zombie attack method.
+	 */
+	@Override
+	public void attack(Board board) {
+		Plant plantTarget = board.getPlant(getRow(), getCol());
+		if(this.getCol()==plantTarget.getCol()+2) {
```
Comment (DwangDistracted) [inline src/assets/Vaulting_Zombie.java]: We have to make sure that the plant vault once

File: src/assets/Vaulting_Zombie.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+			this.setDefaultSpeed();
+			this.setRow(plantTarget.getRow());
+			this.setColumn(plantTarget.getCol());
```
Comment (DwangDistracted) [inline src/assets/Vaulting_Zombie.java]: Isn't the zombie supposed to vault OVER a plant? and not to it?

File: src/assets/Repeater_Peashooter.java (hunk 4 above)
Comment (tanishagarg2) [reply src/assets/Repeater_Peashooter.java]: Repeater peashooter multiplies the attack of a single peashooter by 2.

Comment (DwangDistracted) [reply src/assets/Repeater_Peashooter.java]: adhere to the design docs tanisha

File: src/assets/Repeater_Peashooter.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
+package assets;
+
+import engine.Board;
+import util.Logger;
+
+public class Repeater_Peashooter extends Plant{
+private static Logger LOG = new Logger("Repeated Peashooter");
+	
+	private static final int DEFAULT_HP = HEALTH_MEDIUM;
+	private static final int DEFAULT_POWER = ATTACK_LOW * 2;
```
Comment (DwangDistracted) [inline src/assets/Repeater_Peashooter.java]: Use the provided constants. None of this "+magic number" or "*magic number" stuff

File: src/assets/Vaulting_Zombie.java [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+import engine.Board;
+
+/**
+ * valuting zombie valuts to the first plant in the row and attacks it
+ * @author Tanisha Garg
+ *
+ */
+public class Vaulting_Zombie extends Zombie{
+	private static final int DEAFAULT_SPEED = SPEED_HIGH;
```
Comment (DwangDistracted) [inline src/assets/Vaulting_Zombie.java]: Spelling

File: src/assets/Vaulting_Zombie.java [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+	}
+	
+	/**
+	 * Zombie attack method.
+	 */
+	@Override
+	public void attack(Board board) {
+		Plant plantTarget = board.getPlant(getRow(), getCol());
+		if(this.getRow() == plantTarget.getRow() && this.getCol() == plantTarget.getCol() + 1) {
+			this.setColumn(plantTarget.getCol() - 1);
```
Comment (DwangDistracted) [inline src/assets/Vaulting_Zombie.java]: Can we make sure that this:
1 - Only allows the zombie to jump once
2 - Won't result in the zombie jumping off the board

File: src/assets/Vaulting_Zombie.java [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+			board.onZombieMove(this);
+		}
```
Comment (DwangDistracted) [inline src/assets/Vaulting_Zombie.java]: This zombie isn't able to attack normally anymore

File: src/assets/Jalapeno.java [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+			board.removePlant(row, column);
```
Comment (DwangDistracted) [inline src/assets/Jalapeno.java]: Don't remove this here. See Game for how to remove (reference the potato mine)

File: src/assets/Repeater_Peashooter.java [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+	@Override
+	public void attack(Board board) {
+		int row = getRow();
+		int column = getCol();
+		
+		ArrayList<Zombie> zombieInRow = new ArrayList<>();
+		zombieInRow = (ArrayList<Zombie>) board.getRowTargets(row, column);
+		
+		for(Zombie z : zombieInRow) {
+			if (z != null){
```
Comment (DwangDistracted) [inline src/assets/Repeater_Peashooter.java]: Unnecessary Null check

File: src/assets/Repeater_Peashooter.java [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+	}
+	
+	@Override
+	public void attack(Board board) {
+		int row = getRow();
+		int column = getCol();
+		
+		ArrayList<Zombie> zombieInRow = new ArrayList<>();
+		zombieInRow = (ArrayList<Zombie>) board.getRowTargets(row, column);
+
```
Comment (DwangDistracted) [inline src/assets/Repeater_Peashooter.java]: add null check for zombieInRow here

File: src/assets/Jalapeno.java [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+		}
+		
+		LOG.debug(String.format("Jalapeno at : (%d, %d) defeats all Zombies in row: (%d)", row));
```
Comment (DwangDistracted) [inline src/assets/Jalapeno.java]: not enough arguments for the number of format specifiers

**Pull request merged**